src/main/java/org/scijava/io/handle/DataHandles.java

```diff
@@ -108,4 +111,135 @@ private static synchronized void initUTFMethod() {
+	 * @throws IOException
+	 */
+	public static long copy(final DataHandle<Location> in,
+		final DataHandle<Location> out, final int length) throws IOException
+	{
+		return copy(in, out, length, null);
+	}
+
+	/**
+	 * Copies up to <code>length</code> bytes from the input to the output handle,
+	 * reporting the progress to the provided task. Reading and writing start at
+	 * the current positions of the handles. Stops early if there are no more
+	 * bytes available from the input handle.
+	 *
+	 * @param in input handle
+	 * @param out the output handle
+	 * @param length maximum number of bytes to copy, will copy all bytes if set
+	 *          to <code>0</code>
+	 * @param task a task object to use for reporting the status of the copy
+	 *          operation. Can be <code>null</code> if no reporting is needed.
+	 * @return the number of bytes copied
+	 * @throws IOException
+	 */
+	public static long copy(final DataHandle<Location> in,
+		final DataHandle<Location> out, final long length, final Task task)
+		throws IOException
+	{
+		return copy(in, out, length, task, 64 * 1024);
+	}
+
+	/**
+	 * Copies up to <code>length</code> bytes from the input to the output handle,
+	 * reporting the progress to the provided task. Reading and writing start at
+	 * the current positions of the handles. Stops early if there are no more
+	 * bytes available from the input handle. Uses a buffer of the provided size,
+	 * instead of using the default size.
+	 *
+	 * @param in input handle
+	 * @param out the output handle
+	 * @param length maximum number of bytes to copy, will copy all bytes if set
+	 *          to <code>0</code>
+	 * @param task a task object to use for reporting the status of the copy
+	 *          operation. Can be <code>null</code> if no reporting is needed.
+	 * @return the number of bytes copied
+	 * @throws IOException
+	 */
+	public static long copy(final DataHandle<Location> in,
+		final DataHandle<Location> out, final long length, final Task task,
+		final int bufferSize) throws IOException
+	{
+
+		// get length of input
+		long inputlength;
+		try {
+			inputlength = in.length();
+		}
+		catch (final IOException exc) {
+			// Assume unknown length.
+			inputlength = 0;
+		}
+
+		if (task != null) {
+			if (length > 0) task.setProgressMaximum(length);
+			else if (inputlength > 0) task.setProgressMaximum(inputlength);
+		}
+
+		final byte[] buffer = new byte[bufferSize];
+		long totalRead = 0;
+
+		while (true) {
+			if (task != null && task.isCanceled()) break;
+			final int r;
+			// ensure we do not read more than required into the buffer
+			if (length > 0 && totalRead + bufferSize > length) {
+				int remaining = (int) (length - totalRead);
+				r = in.read(buffer, 0, remaining);
+			}
+			else {
+				r = in.read(buffer);
+			}
+			if (r <= 0) break; // EOF
+			if (task != null && task.isCanceled()) break;
+			out.write(buffer, 0, r);
+			totalRead += r;
+			if (task != null) task.setProgressValue(task
```

Comment:
Should use curly braces for the consequent here, so formatter does the right thing.